gyp/gyp

```diff
@@ -5,4 +5,4 @@
 
 set -e
 base=$(dirname "$0")
-exec python "${base}/gyp_main.py" "$@"
+exec python3 "${base}/gyp_main.py" "$@"
```

Comment:
Shouldn't this variable consider python-related settings described here?
https://github.com/nodejs/node-gyp#configuring-python-dependency

Reply:
[My PR](https://github.com/nodejs/node-gyp/pull/2362) would effectively do what you're suggesting @biodranik. (I haven't re-reviewed my own PR in a while, so I'd want a chance to do that before/if it got merged.)

(For those familiar with the deep workings of this repo: [My PR](https://github.com/nodejs/node-gyp/pull/2362) would make gyp run with the Python binary found by `find-python.js`... which is the script that honors all those Python binary configuration options you linked to. ([these.](https://github.com/nodejs/node-gyp#configuring-python-dependency)))

Granted, most users with a Python binary that can run gyp have it accessible on their PATH as `python` and/or as `python3`. (Or they have _some_ compatibly Python as `python` and/or `python3` on their PATH).

For the few who rely exclusively on those [Python binary configuration options](https://github.com/nodejs/node-gyp#configuring-python-dependency) node-gyp offers to pass `find-python.js`, who _don't have such a Python binary on their PATH, it feels like the configs are a bit of a lie, and them not being used to actually run gyp is a bit of a broken promise implied by the existence of the configs.

Reply:
Actually... digging deeper and doing some quick testing... the file `gyp/gyp` modified by this PR _is apparently never run by node-gyp_. This file is a tiny bash script wrapper around `gyp/gyp_main.py`. node-gyp skips the wrapper in favor of running `gyp/gyp_main.py` directly.

<details>

It does so here:
- https://github.com/nodejs/node-gyp/blob/v9.0.0/lib/configure.js#L244-L245
  - https://github.com/nodejs/node-gyp/blob/v9.0.0/lib/configure.js#L194

```console
$ ~/node-gyp/bin/node-gyp.js configure
gyp info it worked if it ends with ok
gyp info using node-gyp@8.4.1
gyp info using node@16.15.0 | linux | x64
gyp info find Python using Python version 3.8.10 found at "/usr/bin/python3"
gyp info spawn /usr/bin/python3
gyp info spawn args [
gyp info spawn args   '/home/[user]/node-gyp/gyp/gyp_main.py',
[ . . . ]
gyp info spawn args ]
```

I also modified the `gyp/gyp` script to echo some stuff and `touch` a file at an absolute path, then ran `node-gyp rebuild` to see if anything happened. Nothing was echoed from `gyp/gyp`, and the file was not `touch`ed/was not present on my disk. So... I conclude the `gyp/gyp` script was not run by `node-gyp`.

_(end of expanded info)_

</details>

Given the above, **I think this PR is fully unnecessary** for the `node-gyp` repo, should make zero difference, as this is in a file that is not ever being run. (_Might_ be of some use at https://github.com/nodejs/gyp-next??)